dockstore-webservice/src/main/java/io/dockstore/webservice/resources/TokenResource.java

```diff
@@ -378,6 +383,8 @@ public Token addGoogleToken(@ApiParam(hidden = true) @Auth Optional<User> authUs
             }
         }
 
+        acceptTOSAndPrivacyPolicy(user);
```

Comment:
Hmm, this will be called when a user unlinks/relinks their Google account as well; not just during registration/login.

Same applies to GitHub.

This may be OK, and you may not have another (easy) option. Mentioning it in case anybody can think of an issue.

Reply:
If I'm reading correctly, the code below won't override if the values are set already, but it would be worth a test to enforce this behaviour (i.e. test verifies that re-logging in doesn't overwrite the dates)